include/aspect/particle/generator/random_uniform.h

```diff
@@ -0,0 +1,86 @@
+/*
+ Copyright (C) 2011 - 2014 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef __aspect__particle_generator_random_uniform_h
+#define __aspect__particle_generator_random_uniform_h
+
+#include <aspect/particle/generator/interface.h>
+#include <aspect/simulator_access.h>
+
+#include <boost/random.hpp>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Generator
+    {
+      // Generate random uniform distribution of particles over entire simulation domain
+      template <int dim>
+      class RandomUniform : public Interface<dim>, public SimulatorAccess<dim>
+      {
+        public:
+          /**
+           * Constructor.
+           *
+           * @param[in] The MPI communicator for synchronizing particle generation.
+           */
+          RandomUniform();
+
+          /**
+           * Generate a uniformly randomly distributed set of particles in the current triangulation.
+           */
+          // TODO: fix the particle system so it works even with processors assigned 0 cells
+          virtual
+          void
+          generate_particles(Particle::World<dim> &world,
+                             const double total_num_particles);
+
+          /**
+           * Generate a set of particles uniformly randomly distributed within the
+           * specified triangulation. This is done using "roulette wheel" style
+           * selection weighted by cell volume. We do cell-by-cell assignment of
+           * particles because the decomposition of the mesh may result in a highly
+           * non-rectangular local mesh which makes uniform particle distribution difficult.
+           *
+           * @param [in] world The particle world the particles will exist in
+           * @param [in] num_particles The number of particles to generate in this subdomain
+           * @param [in] start_id The starting ID to assign to generated particles
+           */
+          void uniform_random_particles_in_subdomain (Particle::World<dim> &world,
+                                                      const unsigned int num_particles,
+                                                      const unsigned int start_id);
```

Comment:
I suspect that this function could be `private`. Why not incorporate it right away into `generate_particles`?